MatrixSDK/Data/MXRoomSummaryUpdater.m

```diff
@@ -265,6 +266,31 @@ - (void)checkRoomCreateStateEventPredecessorAndUpdateObsoleteRoomSummaryIfNeeded
     }
 }
 
+- (void)checkRoomIsVirtualWithCreateEvent:(MXEvent*)createEvent summary:(MXRoomSummary*)summary session:(MXSession *)session
+{
+    MXRoomCreateContent *createContent = [MXRoomCreateContent modelFromJSON:createEvent.content];
+    
+    if (createContent.isVirtual && [summary.creatorUserId isEqualToString:createEvent.sender])
+    {
+        summary.hiddenFromUser = YES;
+        
+        //  TODO: Below operation can move to a class like 'MXRoomAccountDataUpdater' in future.
```

Comment:
> Did the VoIP team consider MSC1840: Typed rooms to identify virtual rooms?
The first usage of this MSC will be technical, hidden to end-users, rooms for spaces
>
> I am not sure if it will help but I can see all devices of the user doing this setAccountData request.

The team considered it but decided to continue as is.

> Whatever the support of MSC1840, we should declare a room type for those rooms in https://github.com/matrix-org/matrix-ios-sdk/pull/1038/files#diff-eff10c3a41f4d8a73acf293725faa9a78b6a20359d2eff2b6f386cb200956a1aR18 (PR not merged yet)

I'm not sure if we should do that, as `MXRoomType` is tightly coupled with the room create event, however virtual rooms may utilize room create event, but doesn't have to. I think it's better to separate virtual rooms from the MSC1840  room types.